### PR TITLE
Add docker-build package

### PR DESCRIPTION
**Pull request by blampe**
## Adding a new package?

If this pull request adds a new package:

- [x] The package's schema URL is correct.
- [x] The package metadata file, if present, contains: 
  - [x] a supported category (one of `Cloud`, `Infrastructure`, `Network`, `Database`, `Monitoring`, or `Utility`).
  - [x] a description that explains what the package does.
  - [x] a valid logo URL that points to a PNG whose dimensions conform to the others in this repo (e.g., 100x100).
  - [x] a version number prefixed with `v` that corresponds with a valid GitHub release.
- [x] The package repo contains an Overview doc (`/docs/_index.md`) that includes:
  - [x] a brief explanation of what the package is and what it does.
  - [x] at least one representative example in all supported languages.
  - [x] a front-matter property for the `layout` set to `package`.
- [x] The package repo contains an Installation and Configuration doc (`/docs/installation-configuration.md`) that includes:
  - [x] links to SDKs in all supported languages.
  - [ ] a copyable command for installing the resource plugin if necessary.
  - [x] an example of configuring the provider with `pulumi config set`.
  - [x] an example of configuring the provider with environment variables.
- [ ] Someone from the @pulumi/ai team has reviewed the PR.
- [x] Someone from the @pulumi/docs team has reviewed all documentation.

### CODE REVIEW

File: themes/default/content/registry/packages/docker/buildx/--index/_index.md [hunk 1]
```diff
@@ -0,0 +1,1310 @@
...
+<!-- WARNING: this file was generated by Pulumi Docs Generator. -->
+<!-- Do not edit by hand unless you're certain you know what you are doing! -->
+
+An index (or manifest list) referencing one or more existing images.
+
+Useful for crafting a multi-platform image from several
+platform-specific images.
+
+This creates an OCI image index or a Docker manifest list depending on
+the media types of the source images.
```
Comment (AaronFriel) [inline themes/default/content/registry/packages/docker/buildx/--index/_index.md:21]: Is this required to publish multi-platform images with `docker.buildx.Image`?

Comment (blampe) [reply themes/default/content/registry/packages/docker/buildx/--index/_index.md:21]: No, you can publish multi-platform just fine but they don't play well with caching https://github.com/docker/buildx/discussions/1382. If you try to cache a multi-platform image you'll get a warning telling you to use this instead.

File: themes/default/content/registry/packages/docker-build/image/_index.md [hunk 2]
```diff
@@ -0,0 +1,14704 @@
...
+API types are unstable. Subsequent releases _may_ require manual edits
+to your state file(s) in order to adopt API changes.
+
+`retainOnDelete: true` is recommended with this resource until it is
+stable. This enables future API changes to be adopted more easily by renaming
+resources.
+
+Only use this resource if you understand and accept the risks.
+
+## Migrating v3 and v4 Image resources
```
Comment (mjeffryes) [inline themes/default/content/registry/packages/docker-build/image/_index.md:31]: It's probably OK to include this all in the resource docs at launch, but I imagine we'll want to move it to a separate guide at some point so this document can explain the resource without basing it off of the user's familiarity with other resources.

File: themes/default/content/registry/packages/docker-build/image/_index.md [hunk 3]
```diff
@@ -0,0 +1,14704 @@
...
+This is helpful as a safeguard to prevent "broken" images from merging, but users found the behavior unnecessarily redundant when running previews and updates locally.
+
+Version `4.x` changed build-on-preview behavior to be opt-in.
+By default, `v4.x` `Image` resources do _not_ build during previews, but this behavior can be toggled with the `buildOnPreview` option.
+Some users felt this made previews in CI less helpful because they no longer detected bad images by default.
+
+The default behavior of the `Image` resource has been changed to strike a better balance between CI use cases and manual updates.
+By default, Pulumi will now only build `Image` resources during previews when it detects a CI environment like GitHub Actions.
+Previews run in non-CI environments will not build images.
+This behavior is still configurable with `buildOnPreview`.
```
[the remark this answers is not in the document]
Comment (blampe) [reply themes/default/content/registry/packages/docker-build/image/_index.md]: It's not -- I've just updated it to point to the latest release which has updated docs for preview behavior. (We also don't need to check these in.)

File: themes/default/data/registry/packages/docker-build.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+description: "A Pulumi package for building modern Docker images with BuildKit"
+featured: false
+logo_url: "https://www.pulumi.com/logos/pkg/docker.svg"
+name: "docker-build"
+native: true
+package_status: "public_preview"
+publisher: "Pulumi"
+repo_url: "https://github.com/pulumi/pulumi-docker-build"
+schema_file_path: "provider/cmd/pulumi-resource-docker-build/schema.json"
+title: "Docker-Build"
```
Comment (sean1588) [inline themes/default/data/registry/packages/docker-build.yaml:12]:
```suggestion
title: "Docker Build"
```

Docker Build should be fine here with a space, unless that is part of the actual name.

**Review by sean1588 — APPROVED** [review]
thanks for adjusting the index page. lgtm!

**Pull request merged**